src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -369,6 +369,13 @@ private SdkResult ResolveSdkUsingAllResolvers(int submissionId, SdkReference sdk
                     sdkResult = result;
                     return true;
                 }
+                else
```

Comment:
Very minor suggestion:

```suggestion
                else if (loggingContext.LoggingService.MinimumRequiredMessageImportance >= MessageImportance.Low)
```

to avoid the string allocations if there is no logger consuming low-importance messages.